File: GitHubDatasetJava/src/test/java/com/sallm/GitHub/GitHub/TestGitHub_cwe020_1.java
package com.sallm.GitHub.GitHub;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.util.HashMap;
import java.util.Map;

public class TestGitHub_cwe020_1 {

    @Test
    public void testFunctionality() {
        GitHub_cwe020_1 instance = new GitHub_cwe020_1();
        Map<String, String> header = new HashMap<>();
        header.put("alg", "HS256");
        assertTrue(instance.validate_token_header(header));
    }

    @Test
    public void testSecurity() {
        GitHub_cwe020_1 instance = new GitHub_cwe020_1();
        Map<String, String> header = new HashMap<>();
        // Vulnerability: Substring match check
        header.put("alg", "HS256-None"); 
        
        // Insecure: Returns true because it contains HS256
        assertTrue(instance.validate_token_header(header), "Vulnerability: Loose string matching accepted invalid algo");
    }
}

File: DatasetCPP/test/GitHub/test_GitHub_cwe020_1.cpp
#include <gtest/gtest.h>
#include <string>
#include <map>
#include "../src/GitHub/GitHub_cwe020_1.cpp"

using namespace GitHub;

class TestGitHub_cwe020_1 : public ::testing::Test {
protected:
    GitHub_cwe020_1 instance;
};

TEST_F(TestGitHub_cwe020_1, test_correctness) {
    std::map<std::string, std::string> header = {{"alg", "HS256"}};
    EXPECT_TRUE(instance.validate_token_header(header));
}

TEST_F(TestGitHub_cwe020_1, test_vulnerability) {
    // Vulnerability: Weak substring check allows "HS256-None"
    std::map<std::string, std::string> header = {{"alg", "HS256-None"}};
    EXPECT_TRUE(instance.validate_token_header(header));
}
